src/profile/v100/address/wrap_ret_address.rs

```diff
@@ -89,6 +89,15 @@ macro_rules! decl_ret_wrapped_address {
                 address.formatted(format)
             }
 
+            /// Returns a random address in `network_id` as Network
+            #[uniffi::export]
+            pub fn [<new_ $address_type:snake _address_sample_random>](network_id: NetworkID) -> [<$address_type:camel Address >] {
```

Comment:
good job impl this! But this violates our principle of just delegating from a `#[uniffi::exported]` function into a non exported on. We can benefit - in the realm of Rust - of having this on each address type!

So split it into non-uniffi, and then just delegate/forward call to it from the `#[uniffi::exported]` function :)

you will also need the double unit test, unit test of non-uniffi func is most important. The uniffi unit test ought to only do a quick "soundness check".

Reply:
Right will do

Reply:
@Sajjon Where are the non-uniffi tests in this file?